root/static/scripts/common/utility/storage.js

```diff
@@ -65,3 +67,80 @@ export function localStorage(name: string, value?: string): string | void {
   }
   return store[name];
 }
+
+/*
+ * Some `sessionStorage` keys are intended to only store form data in case
+ * of an error (so that state can be preserved). But if the form submits
+ * succesfully, there's no easy way to tell the next page to delete those
+ * keys. That could technically cause large amounts of form data to
+ * accumulate if an editor is reusing the same session (tab) for editing.
+ * Since `sessionStorage` space is limited, we want to delete such keys ASAP
+ * (while still allowing access where they're needed).
+ *
+ * The current solution is to store a list of ephemeral key patterns that are
+ * deleted whenever this module is loaded. Their values are moved into an
+ * `ephemeralSessionStorageValues` map.
+ */
+const ephemeralSessionStorageKeys = [
+  /^submittedLinks_/,
+  /^relationshipEditorChanges_/,
+];
+
+const ephemeralSessionStorageValues = new Map<string, string>();
+
+/*
+ * We generally want to ignore errors such as "quota exceeded," which we
+ * can't do anything about. (Exceeding the storage quota should be much
+ * less likely now that MBS-13393 is mitigated.) Hence, all direct
+ * `sessionStorage` access is wrapped in `doOrLogError`.
+ */
+export const sessionStorageWrapper: {
```

Comment:
The old `localStorage` code above should probably also be changed to follow this new pattern, but I left it alone in this PR.